app/controllers/resources_controller.rb

```diff
@@ -38,8 +38,44 @@ def suggestions
     render json: ToolFilterService.new(params).call, include: params[:include], fields: field_params, status: :ok
   end
 
+  def publish_translation
+    if valid_publish_params?
+      publish_translations
+      head :no_content
+    else
+      render json: {errors: {"errors" => [{source: {pointer: "/data/attributes/id"}, detail: "Record not found."}]}}, status: :unprocessable_entity
+    end
+  end
+
   private
 
+  def valid_publish_params?
+    params["data"] &&
+      params["data"]["relationships"] &&
+      params["data"]["relationships"]["languages"] &&
+      params["data"]["relationships"]["languages"]["data"] &&
```

Comment:
I think you could use `dig` here to make this more succinct.

Reply:
done